src/util/util.cpp

```diff
@@ -662,24 +708,201 @@ bool dirExists(const string& dirName_in)
 		return true;   // this is a directory!
 
 	return false;    // this is not a directory!
+#else 
+	struct stat sb;
+	return stat(dirName.c_str(), &sb) == 0 && S_ISDIR(sb.st_mode);
+#endif
+#endif
+}
+
+#ifndef WIN32
+// mkdir_p for linux from https://gist.github.com/ChisholmKyle/0cbedcd3e64132243a39
+int mkdir_p(const char* dir, const mode_t mode) {
+	char tmp[PATH_MAX_STRING_SIZE];
+	char* p = NULL;
+	struct stat sb;
+	size_t len;
+
+	/* copy path */
+	len = strnlen(dir, PATH_MAX_STRING_SIZE);
+	if (len == 0 || len == PATH_MAX_STRING_SIZE) {
+		return -1;
+	}
+	memcpy(tmp, dir, len);
+	tmp[len] = '\0';
+
+	/* remove trailing slash */
+	if (tmp[len - 1] == '/') {
+		tmp[len - 1] = '\0';
+	}
+
+	/* check if path exists and is a directory */
+	if (stat(tmp, &sb) == 0) {
+		if (S_ISDIR(sb.st_mode)) {
+			return 0;
+		}
+	}
+
+	/* recursive mkdir */
+	for (p = tmp + 1; *p; p++) {
+		if (*p == '/') {
+			*p = 0;
+			/* test path */
+			if (stat(tmp, &sb) != 0) {
+				/* path does not exist - create directory */
+				if (mkdir(tmp, mode) < 0) {
+					return -1;
+				}
+			}
+			else if (!S_ISDIR(sb.st_mode)) {
+				/* not a directory */
+				return -1;
+			}
+			*p = '/';
+		}
+	}
+	/* test path */
+	if (stat(tmp, &sb) != 0) {
+		/* path does not exist - create directory */
+		if (mkdir(tmp, mode) < 0) {
+			return -1;
+		}
+	}
+	else if (!S_ISDIR(sb.st_mode)) {
+		/* not a directory */
+		return -1;
+	}
+	return 0;
 }
+#endif 
 
 bool createDir(const string& dirName)
 {
-	int ret = CreateDirectory(dirName.c_str(), NULL);
-	if (ret == ERROR_PATH_NOT_FOUND)
+#ifdef USE_FILESYSTEM
+	return fs::create_directories(dirName);
+#else
+#ifdef WIN32
+	std::string fixDirName = dirName;
+	fixupPath(fixDirName, FIXUPPATH_SLASH_SKIP, FIXUPPATH_SLASH_REMOVE);
+	int ret = SHCreateDirectoryExA(NULL, dirName.c_str(), NULL);
+	if (ret != ERROR_SUCCESS)
+	{
+		logf("Could not create directory: %s. Error: %i", dirName.c_str(), ret);
+		return false;
+	}
+	return true;
+#else 
+	if (dirExists(dirName))
+		return true;
+
+	int ret = mkdir_p(dirName.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+	if (ret != 0)
 	{
 		logf("Could not create directory: %s", dirName.c_str());
 		return false;
 	}
 	return true;
+#endif
+#endif
+}
+
+void removeDir(const string& dirName)
+{
+#ifdef USE_FILESYSTEM
+	std::error_code e;
+	fs::remove_all(dirName, e);
+#endif
 }
 
+
+void replaceAll(std::string& str, const std::string& from, const std::string& to) {
+	if (from.empty())
+		return;
+	size_t start_pos = 0;
+	while ((start_pos = str.find(from, start_pos)) != std::string::npos) {
+		str.replace(start_pos, from.length(), to);
+		start_pos += to.length();
+	}
+}
+
+
+void fixupPath(std::string& path, FIXUPPATH_SLASH startslash, FIXUPPATH_SLASH endslash)

```

Comment:
This would eventually be removed in favor of something that can join paths that aren't "fixed". I want to get away from needing perfectly formatted paths for things to work. This code will likely be useful in creating a method that can join unfixed paths though.

Reply:
@wootguy imgui has file dialog for c++17 based at filesystem. Possible it can support c++14 also?
We can replace these function to file/directory open dialog, in future ?

Reply:
If it compiles in VS 2019 then I'm fine with adding that in later. VS seems to be slowest to adopt new features.

Reply:
And https://github.com/AirGuanZ/imgui-filebrowser for c++17

Reply:
@wootguy i can't watch notices in github, and can skip some messages from you. You has discord or something like ?

Reply:
Yeah, in discord I'm w00tguy#9718

I sometimes am missing messages here, too.